### PR TITLE
Add fromOption, fromIO and fromTask

### PR DESCRIPTION
**Pull request by mlegenhausen**

### CODE REVIEW

File: src/Observable.ts [hunk 1]
```diff
@@ -9,8 +9,10 @@ import { Monad1 } from 'fp-ts/lib/Monad'
 import { Monoid } from 'fp-ts/lib/Monoid'
 import * as O from 'fp-ts/lib/Option'
 import { pipe, pipeable } from 'fp-ts/lib/pipeable'
-import { combineLatest, EMPTY, merge, Observable, of } from 'rxjs'
+import { combineLatest, EMPTY, merge, Observable, of, defer } from 'rxjs'
 import { map as rxMap, mergeMap } from 'rxjs/operators'
+import * as IO from 'fp-ts/lib/IO'
```
Comment (gcanti) [inline src/Observable.ts]: we can import only the types here

```ts
import { IO } from 'fp-ts/lib/IO'
import { Task } from 'fp-ts/lib/Task'
```

File: src/Observable.ts [hunk 2]
```diff
@@ -38,6 +40,33 @@ export function getMonoid<A = never>(): Monoid<Observable<A>> {
   }
 }
 
+/**
+ * @since 0.6.5
+ */
+export function fromOption<A>(o: O.Option<A>): Observable<A> {
```
Comment (gcanti) [inline src/Observable.ts:46]: what about

```ts
export function fromOption<A>(o: O.Option<A>): Observable<A> {
  return O.isNone(o) ? EMPTY : of(o.value)
}
```

**Review by gcanti — CHANGES_REQUESTED** [review]

**Review by gcanti — APPROVED** [review]

**Comment by gcanti** [comment]
Thanks @mlegenhausen

**Comment by anthonyjoeseph** [comment]
@mlegenhausen Just out of curiosity, is there a reason `fromIO` uses asynchronous context?

For example, it could have been implemented like so:

```ts
export function fromIO<A>(io: IO<A>): Observable<A> {
  return defer(() => rxOf(io()))
}
```

Is there an advantage to using `async` here instead? Is it conceptually more correct?

**Comment by mlegenhausen** [comment]
> Is there an advantage to using async here instead? Is it conceptually more correct?

Don't think so. I think I chose `async` cause it was more inline with the `fromTask` implementation. Without knowing the internals (and the rxjs documentation isn't there of any help) it could be that your implementation could save one event loop iteration when calling the `IO` but this is only a guess.

**Comment by anthonyjoeseph** [comment]
I think you're right. `fromIO` wasn't working for my [game engine's spec](https://github.com/anthonyjoeseph/game-ts/blob/master/test/Render.test.ts) b/c [TestScheduler doesn't work in async contexts](https://github.com/ReactiveX/rxjs/pull/745), but when I rolled my own [fromSyncIO](https://github.com/anthonyjoeseph/game-ts/blob/master/src/util.ts) it worked just fine.

Could this be a desired PR? It does seem mildly more correct since you can raise into asynchronous context with [observeOn(AsyncScheduler)](https://rxjs-dev.firebaseapp.com/api/operators/observeOn#example) but you can't go back to synchronous context from async.

 It probably doesn't matter for most use cases though, especially since the function is so simple, and I certainly don't mind using my own `fromSyncIO`.

**Comment by mlegenhausen** [comment]
> Could this be a desired PR?

Yes definitely.

**Comment by anthonyjoeseph** [comment]
[PR](https://github.com/gcanti/fp-ts-rxjs/pull/43)

**Pull request merged**